### ISSUE TITLE
Is it possible to capture codes for Bluetooth devices?

### ISSUE THREAD

**Issue by wmweis62** [z3t0/Arduino-IRremote#489]
I am doing some volunteer work for Muscular Dystrophy and I am working with a young man who needs voice recognition to drive his hospital bed. I have an ESP8266 ready to go, but I would like to avoid a hard wired solution if I can. The bed has a remote which I only briefly saw and I do not know if it is IR or BT. If it is IR, I have the Audrino program all set to capture the codes. What do I do if it is not IR but instead Bluetooth? Is it possible to capture the codes for the remote buttons if it is a Bluetooth device? I am waiting to hear back from the manufacturer of the remote to see if they can provide me with the codes. I would then provide those codes to Logitech for possible inclusion into their Harmony Hub. As an alternative, I suppose an ESP32 with Bluetooth could be used, but I have not found any sample code showing BTremote libraries, the equivalent of the IRremote libraries.

**Comment by AnalysIR** [comment]
IRremote does not support bluetooth.

However, if the bluetooth device ends up generating IR somewhere, then it should be possible.

I would be surprised if the manufacturer had an 'open' interface on their BT solution. Often, they do not.

Just get someone to look at the remote thru a phone camera and if it is IR, they will see flashes of bright light when looking thru the camera at the remote (close up). If you can get a model number, you may be able to research before visiting.

I guess the other thing to note....is whether the target device can accept IR (regardless of the remote). It could be dual with both IR & BT.

The target device might also have a spare port/connector for control, which you may be able to hack into.

However, for this application I suspect safety, robustness & reliability is paramount. Often the best hack is the simplest. For example, by getting a spare BT remote, it should be possible to open it up and connect directly to the button pads on the inside of the remote(for control by the arduino) & incorporate this 2nd remote into your own project box. This way most complexities are removed.

If the solution is to be deployed in a hospital or similar location, you may also run into other issues with approvals, insurance etc.

**Comment by wmweis62** [comment]
Thanks for the reply. I learned about that trick to point a camera at the LED on a remote after I had the opportunity to see the one for the bed. I will be at their home tomorrow and will know the answer to all this in 24 hours, but as I started thinking about the limitations of someone laying in bed using a remote vs a Bluetooth device, I started thinking I may be facing a Bluetooth device and not a IR device. I was not expecting the IRremote library to help me, but was looking for an equivalent. 

I have considered connecting my ESP8266 into the hand held remote, and perhaps that will end up being my best option, at least for this family. With safety considerations in mind, I modified my program to only execute a bed movement command for 3 seconds. In other words, a Head Up command would run 3 seconds and stop. Due to latency associated with each Google Home command, it made sense to use timed vs separate commands for start and stop. It also reduces the likelihood that I would be driving the same part of the bed in two directions at once. (If I really have to use the approach of a wired connection to the remote, I will add another safety feature which stops all motion before initiating another).

The bigger picture I am trying to help is to define a solution that many families would benefit from. Although it is fun and rewarding to hack something together for a one-off situation, I would find it much more impactful to move this to an out-of-the box experience so many non technical families with a similar need can go buy a Echo or Google Home and a Harmony hub and have a working solution.
